### ISSUE TITLE
Problem with Python TPU code on Windows 10

### ISSUE THREAD

**Issue by wb666greene** [google-coral/edgetpu#162]
I've been off Windows since Window 8 came out, but since the USB TPU acceleration now seems to support Windows 10, a friend gave me a Windows10 PC and asked if I could port my "Security DVR AI Person Detector add-on"  Linux TPU code.

After I installed Python 3.7 and the extra modules that I used,  I started with these instructions for the TPU:
https://coral.ai/docs/accelerator/get-started#1c-on-windows

I installed  edgetpu_runtime_20200629.zip using the batch file and then did:
pip3 install https://dl.google.com/coral/python/tflite_runtime-2.1.0.post1-cp37-cp37m-win_amd64.whl

When I first tried my python code I had an "imp" DLL error which a quick Google search fixed by installing the Visual C runtime.

Now my code starts to run, but when I load the MobileNetSSD-v2_coco downloaded model I get this error:

`[INFO] loading Coral mobilenet_ssd_v2_coco model...
Traceback (most recent call last):
  File "tpu.py", line 1008, in <module>
    main()
  File "tpu.py", line 663, in main
    model = DetectionEngine("mobilenet_ssd_v2\mobilenet_ssd_v2_coco_quant_postprocess_edgetpu.tflite")
  File "C:\Users\MINI PC\AppData\Local\Programs\Python\Python37\lib\site-packages\edgetpu\detection\engine.py", line 73, in __init__
    super().__init__(model_path)
  File "C:\Users\MINI PC\AppData\Local\Programs\Python\Python37\lib\site-packages\edgetpu\basic\basic_engine.py", line 92, in __init__
    self._engine = BasicEnginePythonWrapper.CreateFromFile(model_path)
RuntimeError: Internal: Unsupported data type in custom op handler: 0Node number 0 (edgetpu-custom-op) failed to prepare.
Failed to allocate tensors.
`
Searching found what looks to be the same issue on older Linux versions. but nothing about Windows10.

My code reports edgetpu version as:
Edgetpu_api version: 2.14.0

I'm not directly importing tflite.

Any suggestions as to what is gone wrong here?

**Comment by wb666greene** [comment]
Sorry,  Looks like I had bad luck in timing of my installation.
I had installed edgetpu_runtime_20200624.zip from an older version of the installation page.

My port is running fine after I noticed this.

**Comment by wb666greene** [comment]
I tried again with another fresh Windows 10 system this time starting with edgetpu_runtime_20200629.zip the problem now is that import edgetpu fails with no module.

The python path browser confirmed that  edgetpu was not in site-packages.

I went back to the original system where I initially installed edgetpu_runtime_20200624.zip and copied the two edgetpu folders to the new system.  Now everything works.

Looks like with edgetpu_runtime_20200629.zip  the install.bat file is not installing the edgetpu package.

**Comment by Namburger** [comment]
@wb666greene humnn, thanks for reporting this. 
it is weird indeed that it is not installing the libraries, since this should have taken care of it:
```
copy "%ROOTDIR%\libedgetpu\%FREQ_DIR%\x64_windows\edgetpu.dll" c:\windows\system32
copy "%ROOTDIR%\third_party\libusb_win\libusb-1.0.dll" c:\windows\system32
```
I'm not sure if this is system independence or something is wrong with our installation scripts, could you possibly try executing the `uninstall.bat` and reinstall again with the `edgetpu_runtime_20200629.zip`?

**Comment by wb666greene** [comment]
The uninstall has an error at the end about OEM driver, sorry but I couldn't capture it, (I'mnot much of a Windows user).  It did remove the edgetupu Python packages from site-packages so import edgetpu fails with no module.  

These I had copied manually from the installation I did initially with 20200624.zip which had the unsupported data type error mentioned in the initial post that was fixed by installing 20200629 over it (I didn't notice uninstall.bat in 20200624).

I then downloaded a fresh edgetpu_runtime_20200629.zip, extracted it and ran install.bat.  I got some messages saying that some stuff was already installed, so it seems uninstall.bat is not uninstalling "everything". 

But when the install completed, I started Python and import edgetpu now worked.

Unless there has been a "silent" change to edgetpu_runtime_20200629.zip between now and what I downloaded four days ago, I can't explain it.

I want to point out the libraries (DLL files) were installed to system32, its was the edgetpu Python module that was not installed to site packages (where the tflite Python package was also installed).

**Comment by Namburger** [comment]
@wb666greene I see, sorry i was confused and thought that the dll wasn't there
The `edgetpu` python module should be installed as a pip package from here: https://coral.ai/software#edgetpu-python-api

I guess at last everything is working on your side now? 
Do you think some documentations that we should update to make this more clear?

**Comment by wb666greene** [comment]
Yes everything seems to be working fine on my side now.

I didn't see any reference to  the pip install command in the getting started Windows 10 section. I certainly didn't run it on either system, perhaps 20200624 ran it whereas 20200629 seems not to?

I think the getting started Windows 10 section (https://coral.ai/docs/accelerator/get-started#1c-on-windows) needs to mention that the pip install is needed for Windows to use the edgetpu Python module.

Perhaps its expected to to happen in the parrot example "bash install_requirements.sh" command, but I quickly gave up trying to get git for Windows or Windows Subsystem for Linux working.   

One other inconsistency is in the tables 20200629 says version 2.14.1 but the pip wheels say edgetpu-2.14.0

**Comment by Namburger** [comment]
@wb666greene 
ahh, so the demo on our getting started page will guide you to the [tflite](https://github.com/google-coral/tflite/tree/master/python/examples) api of the code. So it doesn't exactly install the edgetpu module either. 
The examples in this repo will requires the edgetpu module, which users will need to install the pip package from there.
Apologies for all the confusions :(

I totally understand, last time that I used window for a project (a few years back), I had a process running for days. When I came back to check, I realized that process was killed because... auto window updates haha
